### PR TITLE
Updated Neo compilation notebook for GluonCV Yolo example

### PR DESCRIPTION
**Pull request by bpawlik**
Solves #1615

*Description of changes:*
- Pascal VOC URL changed to: https://course.fast.ai/datasets#image-localization
- MXNet framework updated to version 1.7
- Updates necessary to account for SageMaker Neo service level changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py [hunk 1]
```diff
@@ -524,22 +432,31 @@ def model_fn(model_dir):
     :param: model_dir The directory where model files are stored.
     :return: a model (in this case a Gluon network)
     """
+    logging.info('Invoking user-defined model_fn')
+
+    import neomxnet
+
+    #check if GPUs area available
```
Comment (samskalicky) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py:439]: is this comment still valid?

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py [hunk 2]
```diff
@@ -549,9 +466,7 @@ def transform_fn(net, data, content_type, output_content_type):
     x, image = gcv.data.transforms.presets.yolo.transform_test(mx.nd.array(data), 320)
     
     #check if GPUs area available
```
Comment (samskalicky) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py:468]: same thing here, still valid?

**Review by samskalicky — DISMISSED** [review]
Thanks for the update @bpawlik ! Left a few small comments, overall looks good.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb [hunk 3]
```diff
@@ -13,9 +13,10 @@
     "  2. [Convert data into RecordIO](#Convert-data-into-RecordIO)\n",
```
Comment (prats13bag) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:13]: <p>Please change "<strong><em>This is notebook is for demostration purpose only." </em></strong>to<strong><em> "This notebook is for demonstration purpose only."</em></strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1638/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py [hunk 4]
```diff
@@ -524,22 +432,31 @@ def model_fn(model_dir):
...
     return net
     
 def transform_fn(net, data, content_type, output_content_type): 
     """
     Transform incoming requests.
     """
-    import os
-
-    os.system('pip3 install gluoncv')
-
```
Comment (prats13bag) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py:542]: can we log the following at this point:
`logging.info('Invoking user-defined transform_fn')`

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py [hunk 5]
```diff
@@ -524,22 +432,31 @@ def model_fn(model_dir):
     :param: model_dir The directory where model files are stored.
     :return: a model (in this case a Gluon network)
     """
+    logging.info('Invoking user-defined model_fn')
+
+    import neomxnet
```
Comment (prats13bag) [inline sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py:437]: can we instead have the following:
`import neomxnet  # noqa: F401`

the comment aside import will prevent linters to remove the import as this is not being directly used code-wise but it indirectly required and used.

Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py]: Thanks, just added that in the following commit.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py (hunk 4 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py:542]: Done.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb (hunk 3 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/gluoncv_yolo_darknet_neo.ipynb:13]: Good catch, thanks :)

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py (hunk 2 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py]: Thanks, changed the comment in the following commit.

File: sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py (hunk 1 above)
Comment (bartek-aws) [reply sagemaker_neo_compilation_jobs/gluoncv_yolo_darknet/train_yolo.py]: Nope :) Updated that just now.

**Review by prats13bag — APPROVED** [review]
LGTM, approved!
Thanks!

**Review by samskalicky — APPROVED** [review]

**Review by ngluna — APPROVED** [review]

**Review by ngluna — APPROVED** [review]

**Review by ngluna — APPROVED** [review]

**Comment by aaronmarkham** [comment]
CI timeout. Going to bypass.

**Pull request merged**